File: ObjcExample/TestProj/style/SimpleSpinnerViewController.m
//
//  SimpleSpinnerViewController.m
//  TestProj
//
//  Created by Nick on 22/09/21.
//  Copyright © 2021 Nick Wang. All rights reserved.
//

#import "SimpleSpinnerViewController.h"

@interface SimpleSpinnerViewController ()

@property (strong, nonatomic) NestedSpinnerView *spinner;

@end

@implementation SimpleSpinnerViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"Simple Spinner";
    [self setup:@"Dark Style"];
}

- (void)setup:(NSString *)style {
    [self setStyle:style];
    _spinner = [[NestedSpinnerView alloc] initWithFrame:CGRectMake(50, 160, self.view.bounds.size.width - 100, 50)];
    _spinner.anchorTextFont = [UIFont boldSystemFontOfSize:18];
    _spinner.anchorTintColor = UIColor.darkTextColor;
    _spinner.dataSource = @[ [[Entity alloc] init:@"Light Style"], [[Entity alloc] init:@"Dark Style"], [[Entity alloc] init:@"Customised Globally"], [[Entity alloc] initWithName:@"Customised Attributes" withData: @(1)], [[Entity alloc] initWithName:@"Customised AnchorView" withData: @(2)] ];
    _spinner.anchorOffset = CGPointMake(0, _spinner.bounds.size.height);
    
    __weak typeof(self) weakSelf = self;
    _spinner.selectionAction = ^(NSIndexPath * _Nonnull indexPath, NSString * _Nonnull value, id _Nullable userdata) {
        if ([userdata isKindOfClass:[Entity class]]) {
            Entity *entity = (Entity *)userdata;
            if (entity.data == nil) {
                [weakSelf.spinner removeFromSuperview];
                [weakSelf setup:entity.name];
                weakSelf.spinner.anchorValue = entity.name;
            } else {
                switch (entity.data.intValue) {
                    case 1:
                        [weakSelf performSegueWithIdentifier:@"identifierCustomisedSimpleSpinner" sender:nil];
                        break;
                    case 2:
                        [weakSelf performSegueWithIdentifier:@"identifierCustomisedAnchorSpinner" sender:nil];
                        break;
                    default:
                        break;
                }
            }
        }
    };
    [self.view addSubview:_spinner];
}

- (void)setStyle:(NSString *)style {
    if ([style isEqualToString:@"Light Style"]) {
        [NestedSpinnerAppearance light];
    } else if ([style isEqualToString:@"Dark Style"]) {
        [NestedSpinnerAppearance dark];
    } else if ([style isEqualToString:@"Customised Globally"]) {
        [NestedSpinnerAppearance customisedStyle];
    }
}

@end

File: Example/NestedSpinner/Style/SimpleSpinnerViewController.swift
//
//  SimpleSpinnerViewController.swift
//  NestedSpinner
//
//  Created by Nick on 22/09/21.
//  Copyright © 2021 Nick Wang. All rights reserved.
//

import UIKit
import NestedSpinner

class SimpleSpinnerViewController: UIViewController {
    
    var spinner: NestedSpinnerView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "Simple Spinner"
        setup("Dark Style")
    }
    
    func setup(_ style: String) {
        setStyle(style)
        spinner = NestedSpinnerView(frame: CGRect(x: 50, y: 160, width: self.view.bounds.width - 100, height: 50))
        spinner.anchorTextFont = UIFont.boldSystemFont(ofSize: 18)
        spinner.anchorTintColor = .darkText
        spinner.dataSource = [Entity("Light Style"), Entity("Dark Style"), Entity("Customised Globally"), Entity("Customised Attributes", 1), Entity("Customised AnchorView", 2)]
        spinner.anchorOffset = CGPoint(x: 0, y: spinner.bounds.size.height)
        spinner.selectionAction = { indexPath, value, userdata in
            if userdata is Entity {
                let entity = userdata as! Entity
                if entity.data == nil {
                    self.spinner.removeFromSuperview()
                    self.setup(entity.name)
                    self.spinner.anchorValue = entity.name
                } else {
                    switch entity.data {
                    case 1:
                        self.performSegue(withIdentifier: "identifierCustomisedSimpleSpinner", sender: nil)
                        break
                    case 2:
                        self.performSegue(withIdentifier: "identifierCustomisedAnchorSpinner", sender: nil)
                        break
                    default:
                        break
                    }
                }
            }
        }
        self.view.addSubview(spinner)
    }
    
    func setStyle(_ style: String) {
        switch style {
        case "Light Style":
            NestedSpinnerAppearance.light()
            break
        case "Dark Style":
            NestedSpinnerAppearance.dark()
            break
        case "Customised Globally":
            NestedSpinnerAppearance.customisedStyle()
            break
        default:
            break
        }
    }
    
}
